lib/src/shared/message.dart

```diff
@@ -72,17 +73,23 @@ List<int> frame(List<int> rawPayload, [Codec codec]) {
 StreamTransformer<GrpcMessage, GrpcMessage> grpcDecompressor({
   CodecRegistry codecRegistry,
 }) {
-  Codec codec = const IdentityCodec();
+  Codec codec;
   return StreamTransformer<GrpcMessage, GrpcMessage>.fromHandlers(
       handleData: (GrpcMessage value, EventSink<GrpcMessage> sink) {
+    print(value);
     if (value is GrpcData && value.isCompressed) {
+      if (codec == null || codecRegistry.lookup(codec.encodingName) == null) {
```

Comment:
I am a bit confused by this condition. Why do we need to lookup the codec again in the registry if if it is not `null`?

Reply:
oops.